### PR TITLE
feat(kit): use 1 as min segment value in `Date`-related masks

### LINKED ISSUE #148: 🚀 - min segments value for Date-related masks
### Which package(s) are relevant/related to the feature request?

@maskito/kit

### Description

Add min value (01) for date-segments 

Value has to be replaced to 01 if:
1. New segment value consists of only zeros
2. Caret not at start of segment

examples: 

`01|. -> backspace -> 0|1.`
`0|1. -> delete -> 01|.`
`2|0. -> backspace -> |00.`
`|20. -> delete -> 0|1`

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #148

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/kit/src/lib/utils/date/tests/get-date-segment-value-length.spec.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import {getDateSegmentValueLength} from '../date-segment-value-length';
+
+describe('getDateSegmentValueLength', () => {
+    it('short date', () => {
+        expect(getDateSegmentValueLength('mm.yy')).toEqual({day: 2, month: 2, year: 2});
```
Comment (nsbarsukov) [inline projects/kit/src/lib/utils/date/tests/get-date-segment-value-length.spec.ts:5]: `{day: 2, month: 2, year: 2}` is expected behaviour ?

not `{day: 0, month: 2, year: 2}` ?

File: projects/demo-integrations/cypress/tests/kit/date/date-basic.cy.ts [hunk 2]
```diff
@@ -18,6 +18,11 @@ describe('Date', () => {
                 ['12', '12', '12'.length],
                 ['121', '12.1', '12.1'.length],
                 ['1211', '12.11', '12.11'.length],
+                ['0', '0', 1],
+                ['00', '01', '01'.length],
+                ['000', '01.0', '01.0'.length],
+                ['0000', '01.01', '01.01'.length],
+                ['00000000', '01.01.0001', '01.01.0001'.length],
             ] as const;
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/tests/kit/date/date-basic.cy.ts:27]: `2000/0|` => Type `0` => `2000/01|`

https://user-images.githubusercontent.com/35179038/225312813-99e8c063-b84f-4fc4-9245-30c5002128e6.MP4



Should it prevent typing for this case (only for basic typing, for paste/drop etc it okey now) ?

For example, the similar case:
```
`2000/1|` => Type 9 => no changes.
```

Comment (nsbarsukov) [reply projects/demo-integrations/cypress/tests/kit/date/date-basic.cy.ts:27]: @waterplea @splincode guys, what do you think ?

File: projects/demo-integrations/cypress/tests/kit/date/date-basic.cy.ts [hunk 3]
```diff
@@ -176,6 +181,22 @@ describe('Date', () => {
...
+                    .type('{backspace}')
+                    .should('have.value', '01.01.2008')
+                    .should('have.prop', 'selectionStart', '0'.length)
+                    .should('have.prop', 'selectionEnd', '0'.length)
+                    .type('5')
+                    .should('have.value', '05.01.2008')
+                    .should('have.prop', 'selectionStart', '05.'.length)
+                    .should('have.prop', 'selectionEnd', '05.'.length);
+            });
         });
```
[the remark this answers is not in the document]
Comment (nsbarsukov) [reply projects/demo-integrations/cypress/tests/kit/date/date-basic.cy.ts:201]: I understand that it native dates do this.
```ts
new Date('2023.02.31') // Fri Mar 03 2023 00:00:00 GMT+0300 (GMT+03:00)

```

But i try to figure our is it expected or not 🤔

[the remark this answers is not in the document]
Comment (vladimirpotekhin) [reply projects/demo-integrations/cypress/tests/kit/date/date-basic.cy.ts:201]: I thought it was kind of feature) No 🚫 February 30

File: projects/kit/src/lib/masks/date-time/postprocessors/min-max-date-time-postprocessor.ts [hunk 4]
```diff
@@ -27,9 +28,21 @@ export function createMinMaxDateTimePostprocessor({
         const [dateString, timeString] = parseDateTimeString(value, dateModeTemplate);
 
         if (!isDateTimeStringComplete(value, dateModeTemplate, timeMode)) {
+            const parsedDate = raiseSegmentValueToMin(
+                parseDateString(dateString, dateModeTemplate),
```
Comment (nsbarsukov) [inline projects/kit/src/lib/masks/date-time/postprocessors/min-max-date-time-postprocessor.ts:32]: **nitpicking:**

We use `parseDateString` and `parseTimeString` (with the the same args) two times in this file. Inside this `if`-condition and further inside `else`-condition.

Probably, we can move them outside `if/else` (raise it by few lines above).

Comment (nsbarsukov) [reply projects/kit/src/lib/masks/date-time/postprocessors/min-max-date-time-postprocessor.ts:32]: Same case with `createMinMaxDatePostprocessor`

**Review by nsbarsukov — APPROVED** [review]

**Review by nsbarsukov — CHANGES_REQUESTED** [review]
cypress tests fail

**Comment by vladimirpotekhin** [comment]
> cypress tests fail

removing default values from `getSegmentValueLength` revealed a mistake [here](https://github.com/Tinkoff/maskito/pull/197/files#diff-ad74df28c795dc850018350dc1c76d5c46929e35588b0a3164ce8c4bf9373cdaL42) :peka:

**Pull request merged**